aquadoggo/src/materializer/tasks/dependency.rs

```diff
@@ -100,42 +104,44 @@ pub async fn dependency_task(context: Context, input: TaskInput) -> TaskResult<T
                             "Pinned relation list field found containing view id: {}",
                             document_view_id
                         );
-                        next_tasks
-                            .push(get_relation_task(&context, document_view_id.clone()).await?);
+
+                        if let Some(task) =
+                            get_relation_task(&context, document_view_id.clone()).await?
+                        {
+                            next_tasks.push(task);
+                        }
                     }
                 }
             }
             _ => (),
         }
     }
 
+    // Now we check all the "parent" or "inverse" relations, that is _other_ documents pointing at
+    // the one we're currently looking at
+    let mut reverse_tasks = get_inverse_relation_tasks(&context, document.schema_id()).await?;
+    next_tasks.append(&mut reverse_tasks);
+
     // Construct additional tasks if the task input matches certain system schemas and all
-    // dependencies have been reduced.
-    let all_dependencies_met = !next_tasks.iter().any(|task| task.is_some());
+    // dependencies have been reduced
+    let all_dependencies_met = next_tasks.is_empty();
```

Comment:
Now that any "reverse_tasks" are included in the `next_tasks` is this check performing as we'd expect? I don't see a reason not to issue a `schema` task if there are reverse tasks still to work on. As I understand it, we only want to make sure all child/direct relation dependencies are met.

Reply:
Right, when we're looking at a `schema_definition` document that's right, I'll fix this!